src/program/mixins.py

```diff
@@ -39,42 +36,33 @@ def dispatch(self, request, *args, **kwargs):
                 reverse("program:proposal_list", kwargs={"camp_slug": self.camp.slug}),
```

Comment:
we cannot return a HttpResponse from inside setup(), only dispatch() can do that. Either raise an exception or keep this as dispatch()